plugins/modules/vm_portgroup_info.py

```diff
@@ -0,0 +1,252 @@
+              'type': 'DISTRIBUTED_PORTGROUP', 'port_id': pg_id,
+              'port_binding': get_dvs_port_allocation(dvs_pg.config.type),
+              'port_allocation': get_dvs_autoExpand(dvs_pg.config.autoExpand),
+              'network_policy': get_dvs_network_policy(dvs_pg.config.defaultPortConfig.macManagementPolicy),
+              'mac_learning': get_dvs_mac_learning(dvs_pg.config.defaultPortConfig.macManagementPolicy.macLearningPolicy),
+              'teaming_policy': get_teaming_policy(dvs_pg.config.defaultPortConfig.uplinkTeamingPolicy),
+              'port_policy': get_port_policy(dvs_pg.config.policy),
+              'vlan_info': get_vlan_info(dvs_pg.config.defaultPortConfig.vlan)
+              }
+
+        if dvs_pg.config.defaultPortConfig.uplinkTeamingPolicy and \
+                dvs_pg.config.defaultPortConfig.uplinkTeamingPolicy.uplinkPortOrder:
+            pg['active_uplinks'] = dvs_pg.config.defaultPortConfig.uplinkTeamingPolicy.uplinkPortOrder.activeUplinkPort
+            pg['standby_uplinks'] = dvs_pg.config.defaultPortConfig.uplinkTeamingPolicy.uplinkPortOrder.standbyUplinkPort
+
+        return pg
+
+    def get_standard_portgroup_detailed(self, pg_id):
+        pg = self.get_standard_portgroup(pg_id)
+        pg_name = str(pg.summary.name)
+        ret_pg = get_standard_portgroup_vlan_vswitch(pg, pg_name)
+        ret_pg['port_id'] = pg_id
+        ret_pg['type'] = 'STANDARD_PORTGROUP'
+        return ret_pg
+
+    def get_portgroup_of_vm(self):
+        vms_nics = {}
+        # Save a dictionary of portgroup details for reuse
+        pg_map = {}
+        for vm in self.vms:
+            vm_detailed = self.get_vm_detailed(vm_name=vm)
+            vm_nics = []
+            for nic in vm_detailed.nics:
+                nic_details = {
+                    'nic_mac_address': vm_detailed.nics[nic].mac_address,
+                    'nic_mac_type': str(vm_detailed.nics[nic].mac_type),
+                    'nic_type': str(vm_detailed.nics[nic].type)
+                }
+
+                pg_type = str(vm_detailed.nics[nic].backing.type)
+                pg_id = str(vm_detailed.nics[nic].backing.network)
+
+                if pg_type not in ['DISTRIBUTED_PORTGROUP', 'STANDARD_PORTGROUP']:
+                    continue
+                if pg_id not in pg_map:
+                    if pg_type == 'STANDARD_PORTGROUP':
+                        pg_map[pg_id] = self.get_standard_portgroup_detailed(pg_id)
+                    else:
+                        pg_map[pg_id] = self.get_dvs_portgroup_detailed(pg_id)
+
+                nic_details.update(pg_map[pg_id])
+                vm_nics.append(nic_details)
+
+            vms_nics[vm_detailed.name] = vm_nics
+
+        return vms_nics
+
+    def get_vm_detailed(self, vm_name):
+        vm_id = self.vmware_client.get_vm_by_name(vm_name)
+        return self.vmware_client.api_client.vcenter.VM.get(vm=vm_id)
+
+    def _vvars(self, vmware_obj):
```

Comment:
This method is reused also in other modules, can we have it in `module_utils`?

Reply:
You're right but I don't really need this function anymore... I removed it from the module